### PR TITLE
Only update merge_base in checker if not already merged

### LINKED ISSUE #10766: [Bug] Merged PRs have their commits and files changed metadata 'zeroed'... sometimes.
- Gitea version (or commit ref): v1.11.3
- Git version: 2.24.1
- Operating system: Ubuntu 18.04 (used as a Docker host for Gitea)
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [x] No
  - [ ] Not relevant
- Log gist: N/A

## Description
As of recent (within the last month I'd estimate), many PRs have their commits and files changed counters zeroed out after merging. This happens most of the time _but weirdly not always_ on my instance now. Deleting the branch afterwards doesn't make seem to make a difference if that makes a difference.

I'm not sure how a lot of the internal Git plumbing works for Gitea so I haven't managed to take a look at the code directly. I know that #10618 seemed to be somewhat relevant - and the linked PR example in there does indeed show the behaviour below! I'm confused as to why this is shown for me as both branches definitely still exist.

## Screenshots
image

### PR DESCRIPTION
**Pull request by zeripath**
Fix #10766

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

**Review by 6543 — APPROVED** [review]

File: services/pull/pull.go [hunk 1]
```diff
@@ -120,7 +120,7 @@ func ChangeTargetBranch(pr *models.PullRequest, doer *models.User, targetBranch
 	if pr.Status == models.PullRequestStatusChecking {
 		pr.Status = models.PullRequestStatusMergeable
 	}
-	if err := pr.UpdateCols("status, conflicted_files, base_branch"); err != nil {
+	if err := pr.UpdateCols("merge_base, status, conflicted_files, base_branch"); err != nil {
```
Comment (zeripath) [inline services/pull/pull.go]:
```suggestion
	if err := pr.UpdateColsIfNotMerged("merge_base, status, conflicted_files, base_branch"); err != nil {
```

**Review by 6543 — APPROVED** [review]

**Review by guillep2k — APPROVED** [review]

File: services/pull/pull.go [hunk 2]
```diff
@@ -120,7 +120,7 @@ func ChangeTargetBranch(pr *models.PullRequest, doer *models.User, targetBranch
 	if pr.Status == models.PullRequestStatusChecking {
 		pr.Status = models.PullRequestStatusMergeable
 	}
-	if err := pr.UpdateCols("status, conflicted_files, base_branch"); err != nil {
+	if err := pr.UpdateColsIfNotMerged("merge_base, status, conflicted_files, base_branch"); err != nil {
```
Comment (guillep2k) [inline services/pull/pull.go]: But "merge_base" is not updated in this function.

Comment (zeripath) [reply services/pull/pull.go]: ah but it is... In TestPatch...

Comment (lafriks) [reply services/pull/pull.go]: For consistency
```suggestion
	if err := pr.UpdateColsIfNotMerged("merge_base", "status", "conflicted_files", "base_branch"); err != nil {
```

**Review by lafriks — APPROVED** [review]

**Pull request merged**